### ISSUE TITLE
blastdbcmd broken when used with several databases

### ISSUE THREAD

**Issue by FredericBGA** [peterjc/galaxy_blast#116]
https://github.com/peterjc/galaxy_blast/blob/0a172d2516a9d7e8d8fa26dff24c0c906b9d0c6a/tools/ncbi_blast_plus/ncbi_blastdbcmd_wrapper.xml#L11

this line should be updated to:

> blastdbcmd -dbtype $db_opts.db_type -db "${db_opts.database.fields.path.replace(',',' ')}"

as blastdbcmd needs a space separated list of databases.

**Comment by peterjc** [comment]
Are you using multi-database entries in the ``*.loc`` file here? What does the entry there look like (spaces, commas)?

**Comment by FredericBGA** [comment]
No, I have only one entry per line in .loc

**Comment by peterjc** [comment]
Oh - I see what you mean now - in the UI you are picking multiple databases from the list populated from the ``*.loc`` file, and Galaxy is building a comma separated string. Do you want to submit this as a pull request?

Note elsewhere in ``ncbi_macros.xml`` I'm using this:

```
'${" ".join(str($db_opts.database.fields.path).split(","))}'
```

The replace is simpler, so you could change that too.

(I'd have to dig though the Galaxy parameter code to see if there is a more elegant approach, but there is always a risk it might break on older releases of Galaxy)

**Comment by FredericBGA** [comment]
:+1: 
I will see if I have time for the PR...

**Comment by FredericBGA** [comment]
this one (in your macro) is fine no?
> '${" ".join(str($db_opts.database.fields.path).split(","))}'
Galaxy will give a comma separated string that will be split then join using space.

**Comment by peterjc** [comment]
#117 merged, leaving this open until I've done a release with the fix.

**Comment by peterjc** [comment]
Looks like I forgot to do the release. Also we don't have a test with multiple databases...

**Comment by peterjc** [comment]
Tests added in #132.

**Comment by peterjc** [comment]
Released to Test Tool Shed as version 2.7.1+galaxy0 - would upload to Main Tool Shed now, but that's giving me an internal server error.

**Comment by peterjc** [comment]
Closing, that's been released now: https://toolshed.g2.bx.psu.edu/repository?repository_id=1d92ebdf7e8d466c

Thank you!
